component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandler.java

```diff
@@ -45,128 +41,89 @@
-        String tenantAwareUsername;
-        String tenantDomain;
-        Map<String, String> claims = new HashMap<>();
+        List<String> generatedBackupCodes = new ArrayList<>();
         try {
-            tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
-            tenantDomain = MultitenantUtils.getTenantDomain(username);
-
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
             if (userRealm != null) {
-                Map<String, String> userClaimValues = userRealm.getUserStoreManager()
-                        .getUserClaimValues(tenantAwareUsername, new String[]{BACKUP_CODES_CLAIM}, null);
-                storedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
-                if (StringUtils.isEmpty(storedBackupCodes) || refresh) {
-                    String plainBackupCodes = BackupCodeUtil.generateBackupCodes(tenantDomain);
-                    if (StringUtils.isNotEmpty(plainBackupCodes)) {
-                        generatedBackupCodes = BackupCodeUtil.encrypt(plainBackupCodes);
-                    }
-                } else {
-                    generatedBackupCodes = storedBackupCodes;
-                }
-                claims.put(BACKUP_CODES_CLAIM, generatedBackupCodes);
-                claims.put(BACKUP_CODES_ENABLED_CLAIM, "true");
+                generatedBackupCodes = BackupCodeUtil.generateBackupCodes(tenantDomain);
             }
+            updateUserProfile(generatedBackupCodes, username);
         } catch (UserStoreException e) {
             throw new BackupCodeException(ERROR_CODE_ERROR_ACCESS_USER_REALM.getCode(),
                     String.format(ERROR_CODE_ERROR_ACCESS_USER_REALM.getMessage(), username, e));
-        } catch (CryptoException e) {
-            throw new BackupCodeException(ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getCode(),
-                    ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getMessage(), e);
         }
-        return claims;
+        return generatedBackupCodes;
     }
 
     /**
-     * Update backup code claims and return backup codes for the user.
+     * Update backup code claims for the user.
      *
-     * @param claims   Map with the backup code claims.
+     * @param generatedBackupCodes Generated backup codes as a comma separated string.
      * @param username Username of the user.
-     * @return Backup codes for the user.
      * @throws BackupCodeException If an error occurred while updating backup codes.
      */
-    public static List<String> updateBackupCodes(Map<String, String> claims, String username)
+    private static void updateUserProfile(List<String> generatedBackupCodes, String username)
             throws BackupCodeException {
 
-        String tenantAwareUsername;
-        String encryptedBackupCodes = claims.get(BACKUP_CODES_CLAIM);
         try {
+            String tenantAwareUsername;
```

Comment:
Can't we move this inside the if block? Or do we need this variable at all?

Reply:
addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/fadf64b3e70925c84689808a42366845bae0660a](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/fadf64b3e70925c84689808a42366845bae0660a)